sandbox/public/index.html

```diff
@@ -50,7 +50,13 @@
         // errorsEnabled: false,
         logEnabled: true,
         prehidingStyle: ".personalization-container { opacity: 0 !important }",
-        idSyncContainerId: 81
+        idSyncContainerId: 81,
+        onBeforeEventSend(xdm) {
```

Comment:
How do you see this changing when we start supporting `data` on the event command? Would there be any reason for a customer to access/modify that data here? If so, would `data` just be a second argument? Would it be better to make the signature `onBeforeEventSend({ xdm, data })`?

Would we need to support a use case where a user wants to always send their data layer on every event, so rather than passing it through each event command, they might just configure `onBeforeEventSend` to just send it every time? If so, how would they do that if their data layer were not XDM? For example:

```
onBeforeEventSend(xdm, data) {
  // I can't do "return dataLayer" because we don't support return values, therefore data would need to be an empty object to which I would need to copy properties from my data layer.
  Object.assign(data, dataLayer);
}
```

Anyway, I realize I was the one that said not to accept return values and still feel like that's probably the right way to go as long as this isn't a use case we need to consider. We can add support for return values later if we need to.

Reply:
I agree that we should make the signature `{xdm, data}`.

Let's start without the return and see how it goes.